### PR TITLE
Add logger namespaces and subnamespaces

### PR DESCRIPTION
**Pull request by ghoz**
availability
home assistant
ota
zigbee
mqtt subnamespaces: connection send receive

* Changed some logger strings to remove redundant information like 'Zigbee' : might need to rework your stored greps if you have any
* need some input on the naming eg : z2m:mqtt:cnx vs z2m:mqtt:connection
* for mqtt, I was torn betwen letting the errors in the "main" z2m:mqtt namespace or adding them to the subnamespace

### CODE REVIEW

**Comment by ghoz** [comment]
@Nerivec pinging you again as this is logger related...
biggest use case for me is filtering out the chatty mqtt publish in info mode
the rest is more as a convenience during debug.

File: lib/mqtt.ts [hunk 1]
```diff
@@ -7,6 +7,9 @@ import bind from 'bind-decorator';
 import type {QoS} from 'mqtt-packet';
 
 const NS = 'z2m:mqtt';
+const NS_CNX = NS+':cnx';
```
Comment (Nerivec) [inline lib/mqtt.ts:10]:
```typescript
const NS_CONNECT = `${NS}:connect`;
```
?

**Comment by Nerivec** [comment]
@Koenkk This introduces namespaces for only some of the z2m classes (previous only had `z2m:mqtt` for the MQTT publish case). We had yet to go over namespaces for z2m, not sure if/how you want to approach this. Though if namespaces are to be introduced for each class, we might want to remove the param default (`z2m`), to ensure we didn't miss any occurrence. _This also splits up mqtt class into several namespaces which departs from the usual pattern._

If desired, I'd go with something like:
`z2m:ext:xyz` for `lib/extension/xyz.ts`
(while respecting `[a-z0-9]`)

I'm a bit worried with the fact that misunderstanding this, could lead users to filter out namespaces and lose frontend main feedback (`z2m` notifies), which could lead to a lot of undesired issues being opened (buttons seemingly not working & such). With the special case of `z2m:mqtt` being supported as a filter, the rest is pretty quiet overall, making other `z2m:?` filters more "dangerous" than necessary in my opinion.

**Comment by ghoz** [comment]
Except for the mqtt one, that will have a positive impact to regular users, 
I don't have a strong opinion on the rest. they usually only output debug or errors and are indeed on the quiet side.
I added them as I had some issues I tried to debug :
* a homeassistant discovery config issue (https://github.com/Koenkk/zigbee2mqtt/issues/22976,  https://github.com/Koenkk/zigbee2mqtt/issues/22977 )
* a device that is pinged while disabled (I think I know where to fix that, need to test it out)
* some devices that request an ota every 5 minutes while there is none available. (need to find my sniffer dongle for this one)
* one crazy device that sends 5 times the same information, sometimes refuses to answer, errors out and refuses to leave the network (hammer time?)

But I agree it would male sense to generalize the namespaceing, or not ...
I did looking at the other extensions but they did not seem to be very chatty at all at least  on my system:
summary of 4h  ~30Mb of debug logs on my system :
```
  88552 debug:  zh:zstack:unpi:parser:
  27311 debug:  zh:zstack:znp:
  10380 debug:  zh:controller:
   9680 debug:  z2m:zigbee:
   8996 info:   z2m:mqtt:send:
   6153 debug:  zh:zstack:
   4880 debug:  zh:controller:endpoint:
   4648 debug:  zh:zstack:unpi:writer:
   1038 debug:  z2m:ota:
    403 debug:  zh:controller:device:
    119 debug:  z2m:
     21 debug:  z2m:mqtt:receive:
     18 debug:  z2m:availability:
     15 debug:  zhc:lumi:
     14 debug:  zh:controller:database:
     10 error:  zh:controller:device:
      9 warning:        z2m:availability:
      9 info:   z2m:
      8
      6 debug:  zhc:light:
      3 error:  z2m:
      1 info:   zhc:tz:
```

**Comment by ghoz** [comment]
as for the "confused users" missing notifications, as the actual namespaces are not documented, and only visible in the logs themselves, I guess we are safe from the casual misclick change ?

File: lib/mqtt.ts (hunk 1 above)
Comment (ghoz) [reply lib/mqtt.ts]:
```ts
const NS_CONNECT = `${NS}:connect`; 
```
!

**Comment by Koenkk** [comment]
@ghoz had some discussion with @Nerivec but we think it's best to only use `NS` in the MQTT receive and send (and remove the other occurrences from MQTT and all the other classes). This should address your use case:

> biggest use case for me is filtering out the chatty mqtt publish in info mode

**Comment by ghoz** [comment]
Understood, closing the PR then.

**Pull request closed without merging**